fix(linter/plugins): handle plugin names containing slashes

overlookmotel opened the pull request:
Fixes #14557.

Split full rule names into plugin name + rule name on *last* slash instead of first slash, if rule name starts with `@`.

e.g. `@tanstack/query/exhaustive-deps`:

* Plugin name: `@tanstack/query`
* Rule name: `exhaustive-deps`

The PR only affects parsing rule names in configs. I'm leaving dealing with `// oxlint-disable @tanstack/query/exhaustive-deps` to a later PR.

overlookmotel commented on crates/oxc_linter/src/config/rules.rs:
```diff
@@ -249,7 +249,12 @@ impl<'de> Deserialize<'de> for OxlintRules {
+    // For scoped packages (starting with `@`), split at the last `/` to handle
+    // packages like `@eslint-react/naming-convention` with rule `rule-name`.
+    // For non-scoped packages, split at the first `/`.
+    let parts = if name.starts_with('@') { name.rsplit_once('/') } else { name.split_once('/') };
```
Fixed.

camc314 commented:
### Merge activity

* **Dec 18, 9:37 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 18, 9:40 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
fix(linter/plugins): handle plugin names containing slashes (#17073)

Fixes #14557.

Split full rule names into plugin name + rule name on *last* slash instead of first slash, if rule name starts with `@`.

e.g. `@tanstack/query/exhaustive-deps`:

* Plugin name: `@tanstack/query`
* Rule name: `exhaustive-deps`

The PR only affects parsing rule names in configs. I'm leaving dealing with `// oxlint-disable @tanstack/query/exhaustive-deps` to a later PR.
